common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -135,6 +129,26 @@ void extractState(@NonNull final Bundle state) {
         mPostPageLoadedJavascript = state.getString(POST_PAGE_LOADED_URL);
         webViewZoomEnabled = state.getBoolean(WEB_VIEW_ZOOM_ENABLED, true);
         webViewZoomControlsEnabled = state.getBoolean(WEB_VIEW_ZOOM_CONTROLS_ENABLED, true);
+        mClientCertAuthChallengeHandler = extractClientCertAuthHandler(state);
+    }
+
+    @SuppressWarnings("unchecked")
+    private IChallengeHandler<ClientCertRequest, Void> extractClientCertAuthHandler(@NonNull final Bundle state){
+        final String challengeHandlerClassName = state.getString(CLIENT_CERT_REQUEST_CLASS_NAME, null);
+        if (!StringUtil.isNullOrEmpty(challengeHandlerClassName)) {
+            try {
+                final Class<?> challengeHandlerClass = Class.forName(challengeHandlerClassName);
+                mClientCertAuthChallengeHandler = (IChallengeHandler<ClientCertRequest, Void>) challengeHandlerClass.newInstance();
+            } catch (ClassNotFoundException e) {
+                // Log. Do nothing
```

Comment:
on second thought.. probably better to throw. we want to make sure any "dev errors" like this are caught during test/bugbash.